GitVersionCore/GitVersionContext.cs

```diff
@@ -11,12 +11,12 @@ public class GitVersionContext
     {
         readonly Config configuration;
 
-        public GitVersionContext(IRepository repository, Config configuration, bool isForTrackingBranchOnly = true)
-            : this(repository, repository.Head, configuration, isForTrackingBranchOnly)
+        public GitVersionContext(IRepository repository, Config configuration, bool isForTrackingBranchOnly = true, string commitId = null)
```

Comment:
To me, Config is something you read from the yaml file. This is not something you want to read from the yaml file (because it's too dynamic). I can easily move it to the config class though, just let me know what you prefer.